source/channel.c

```diff
@@ -805,27 +814,57 @@ struct aws_io_message *aws_channel_slot_acquire_max_message_for_write(struct aws
     return aws_channel_acquire_message_from_pool(slot->channel, AWS_IO_MESSAGE_APPLICATION_DATA, size_hint);
 }
 
-int aws_channel_slot_increment_read_window(struct aws_channel_slot *slot, size_t window) {
+static void s_window_update_task(struct aws_channel_task *channel_task, void *arg, enum aws_task_status status) {
+    (void)channel_task;
+    struct aws_channel *channel = arg;
 
-    if (slot->channel->channel_state < AWS_CHANNEL_SHUTTING_DOWN) {
-        size_t temp = slot->window_size + window;
-        if (temp < slot->window_size) {
-            slot->window_size = SIZE_MAX;
-        } else {
-            slot->window_size = temp;
+    if (status == AWS_TASK_STATUS_RUN_READY && channel->channel_state < AWS_CHANNEL_SHUTTING_DOWN) {
+        /* get the right-most slot to start the updates. */
+        struct aws_channel_slot *slot = channel->first;
+        for (;;) {
+            if (slot->adj_right) {
+                slot = slot->adj_right;
+            } else {
+                break;
+            }
         }
```

Comment:
Super duper trivial, but this could be simpler looking:
```
struct aws_channel_slot *slot = channel->first;
while(slot->adj_right) {
    slot = slot->adj_right;
}
```